geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: second-workload-listeners
+    kind: ListenerSet
+    sectionName: second
+```
+
+For instance, the following `HTTPRoute` attemps to attach to a listener defined in the parent `Gateway`. This is not valid and the route's status should reflect that.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: some-workload-listeners
+    kind: ListenerSet
+    sectionName: foo
+```
+
+#### Listener Validation
+
+Implementations MUST treat the parent `Gateway` as having the concatenated list of all listeners from itself and attached `ListenerSets`
+Validation of this list of listeners MUST behave the same as if the list were part of a single `Gateway`.
+
+From the earlier example the above resources would be equivalent to a single `Gateway` where the listeners are collapsed into a single list.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+
+#### Listener Precedence
+
+Listeners should be merged using the following precedence:
+1. "parent" Gateway
+2. ListenerSet ordered by creation time (oldest first)
+3. ListenerSet ordered alphabetically by “{namespace}/{name}”.
+
+If there are listener conflicts, this should be reported as `Conflicted=True` in the ListenerStatus as usual. See 'Conditions' section below for more details on object should report the conflict.
+
+###  Gateway Conditions
+
+`Gateway` currently supports the following top-level condition types: `Accepted` and `Programmed`
+
+For a `Gateway`, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a `ListenerSet` listener is invalid, `ListenersNotValid` would be reported. 
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
```

Comment:
I think that this is not sufficient, we need to be specific about what both `Accepted` and `Programmed` mean in this case.

Also note that if we make Listeners optional (as per the above discussion), this changes the meaning of `Accepted` again - because now it needs to consider the "no Listeners and no attached ListenerSets" case.

In general, `Accepted` is a relatively local Condition that doesn't make any representations about child objects, so I'd be reluctant to change this here. I think that `Accepted` should mean "this resource as it stands is okay" _only_.

`Programmed` still means "The whole set of resources produces enough valid configuration that _some_ dataplane config has been generated and sent to the dataplane and will be available some time soon." So it implies that it can't be `true` unless there's at least one valid Listener, with valid Routes attached _somewhere_. So I don't think that `Programmed` should be set unless the above is true.

Practically here, in the case that `AllowedListeners` is set, we will need another top-level Condition, something like `AttachedListeners`, that is set to `true` when at least one child Listener arrives from a ListenerSet (remembering that ListenerSets may also have partial validity or may conflict with each other as per the below ListenerSet status discussion). It would be set to `false` when AllowedListeners is set but has no valid listeners attached, and would be absent or present and set to `unknown` in the case that no AllowedListeners config is present.

Reply:
Updated to state `Accepted` & `Programmed` remain unchanged and introduced a new condition.